utils/relationship.js

```diff
@@ -147,6 +147,10 @@ function prepareRelationshipForTemplates(entityWithConfig, relationship, generat
     relationshipNameCapitalized: _.upperFirst(relationshipName),
     relationshipNameHumanized: _.startCase(relationshipName),
     columnName: generator.getColumnName(relationshipName),
+    columnNamePrefix:
+      generator.jhipsterConfig.backwardCompatibleDerivedName && relationship.id && relationship.relationshipType === 'one-to-one'
```

Comment:
This param can be used by blueprints to specify how they would like to name relationships for one-to-one derived ids